degirum_tools/line_count.py

```diff
@@ -154,79 +189,70 @@ def analyze(self, result):
             return
 
         new_trails = set(result.trails.keys())
+        new_trails_list = [new_trails for _ in self._counted_trails_list]
         if self._count_first_crossing:
-            # remove old trails, which are not active anymore
-            self._counted_trails = self._counted_trails & new_trails
-            # obtain a set of new trails, which were not counted yet
-            new_trails -= self._counted_trails
+            for i in range(len(self._counted_trails_list)):
+                # remove old trails, which are not active anymore
+                self._counted_trails_list[i] = (
+                    self._counted_trails_list[i] & new_trails_list[i]
+                )
+                # obtain a set of new trails, which were not counted yet
+                new_trails_list[i] = new_trails_list[i] - self._counted_trails_list[i]
 
         def count_increment(trail_vector, line_vector):
-            counts = SingleLineCounts()
+            increment_counts = self._count_type.__bases__[0]()

```

Comment:
May be instead of using such obscure syntax, just define AND return separate `increment_counts` in each branch of `if`?